### DISCUSSION TITLE
Guide of Adding a New Language (Khmer).

### DISCUSSION THREAD

**Discussion by seanghay** [meilisearch/product#316] [Feedback & Feature Proposal]
I use MeiliSearch with **Khmer `km`** language (Cambodian Language). 
The language doesn't have whitespace the same way as Thai and Lao.

I also happened to build a website for splitting Khmer words using WebAssembly.  

**Project**: **https://github.com/seanghay/khmer-text-segmenter**

**Stolen the idea from**: https://github.com/surferseo/intl-segmenter-polyfill (it uses C, so I think it's compatible with Rust) `break_iterator.c`

**Website**: https://khmer-zwsp.netlify.app/

<img src=https://user-images.githubusercontent.com/15277233/143624940-8bf8c233-55ae-436b-9be8-c1363bab1b1a.png width=500>

**km**: ខ្ញ៉ំស្រលាញ់សត្វព្រៃ 
**en**: I love wild animals

[Intl.Segmenter (It supports Khmer Language)](https://github.com/tc39/proposal-intl-segmenter) is available in the latest Chrome and Node 16.

**Comment by gmourier** [comment]
Hi @seanghay! 👋

Thanks for opening up this discussion. If I understand correctly you have developed in wasm, the tokenization of Khmer?

I guess you would like us to support it on the MeiliSearch side 😇  but I may be wrong.

I would be glad to have more information. Thanks in advance!

**Comment by gmourier** [comment]
Hi @seanghay 👋

I'm locking this discussion in favor of #374

Thanks!
